pkg/apis/mattermost/v1alpha1/clusterinstallation_utils.go

```diff
@@ -53,10 +53,39 @@ func (mattermost *ClusterInstallation) SetDefaults() error {
 
 	mattermost.Spec.Minio.SetDefaults()
 	mattermost.Spec.Database.SetDefaults()
+	err := mattermost.Spec.BlueGreen.SetDefaults(mattermost)
+	if err != nil {
+		return err
+	}
 
 	return nil
 }
 
+// SetDefaults sets the missing values in BlueGreen to the default ones
+func (bg *BlueGreen) SetDefaults(mattermost *ClusterInstallation) error {
+	if bg.Enable == true {
+		if bg.GreenInstallationName == "" {
+			bg.GreenInstallationName = fmt.Sprintf("%s-green", mattermost.Name)
+		}
+		if bg.BlueInstallationName == "" {
+			bg.BlueInstallationName = fmt.Sprintf("%s-blue", mattermost.Name)
+		}
+		if bg.GreenIngressName == "" {
+			bg.GreenIngressName = fmt.Sprintf("green.%s", mattermost.Spec.IngressName)
+		}
+		if bg.BlueIngressName == "" {
+			bg.BlueIngressName = fmt.Sprintf("blue.%s", mattermost.Spec.IngressName)
+		}
+		if bg.GreenVersion == "" || bg.BlueVersion == "" {
+			return errors.New("Both Blue and Green deployment versions required, but not set")
+		}
+		if bg.ProductionDeployment == "" {
```

Comment:
We should check for more than empty here. If the value is neither blue or green, we should error or do something.